backend/src/bundles/users/user.controller.ts

```diff
@@ -167,6 +186,53 @@ class UserController extends BaseController {
             payload: user,
         };
     }
+
+    private async updateUser(
+        options: ApiHandlerOptions<{
+            user: UserAuthResponseDto;
+            body: UserUpdateProfileRequestDto;
+            params: { userId: string };
+        }>,
+    ): Promise<ApiHandlerResponse> {
+        const { user, body, params } = options;
+        const userId = params.userId;
+
+        try {
+            if (Number(userId) !== Number(user.id)) {
+                throw new Error('Token mismatch');
+            }
+            const updatedUser = await this.userService.update(
+                { id: Number(userId) },
+                body,
+            );
+            if (updatedUser && body.dateOfBirth) {
+                const [day, month, year] = body.dateOfBirth.split('/');
+                const parsedDate = new Date(`${year}-${month}-${day}`);
+
+                if (Number.isNaN(parsedDate.getTime())) {
+                    throw new HttpError({
+                        message: UserValidationMessage.BIRTHDATE_FORMAT,
+                        status: HttpCode.BAD_REQUEST,
+                    });
+                } else {
+                    const formattedDate =
+                        parsedDate.toLocaleDateString('en-GB');
+
+                    updatedUser.dateOfBirth = formattedDate;
+                }
+            }
```

Comment:
Why do we need all this logic here? Can't we simply pass the date from the frontend as an ISO string, return it as an ISO string as well, and then just change the format when displaying the date on the frontend?

Reply:
So, when user inputs dateOfBirth I should made a validation of it to be format "dd/mm/yyyy" as was told in specs, so the type of `dateOfBirth `is string in `UserUpdateProfileRequestDto`, but in database it is Date type, so I was converting  it to Date on backend. Or how should I handle it correctly?)